lua/rustaceanvim/server_status.lua

```diff
@@ -7,8 +7,13 @@ local _ran_once = {}
 
 ---@param result rustaceanvim.internal.RAInitializedStatus
 function M.handler(_, result, ctx, _)
-  if result and result.health and result.health ~= 'ok' then
-    vim.notify_once("rust-analyzer health is not OK. Run 'RustLsp logFile' for details", vim.log.levels.WARN)
+  if result and result.health and result.health ~= 'ok' and result.quiescent then
```

Comment:
question: can we move this to after the check on line 19 below? 
I don't think we need to notify more than once per client ID, as you'll likely start a new client when retrying.

If we don't deduplicate somehow, we could risk spamming warning notifications (rust-analyzer can send multiple server status updates with `result.quiescent = true`).
I don't have a reliable way to test this theory though.

Reply:
I never saw a new client ID appear, so the message only displayed once. If I fixed the issue, and then introduced a new one, I never saw the message again.

The way I'm testing this is opening up a Cargo.toml, go to a dependency specified via a path, and change the path to something non-existent. Each time I save I see the message, and if I fix it and break it again, the message reappears.

Reply:
I see. Then let's split the check below into separate `if not quiescent then return` and `if _ran_once[ctx.client_id] then return` checks and move this logic between them.